[package] autoconf/2.71: unable to build test_package with compiler=msvc in profile

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **autoconf/2.71**
  * Operating System+version: **Windows 10 21H2**
  * Compiler+version: **Visual Studio 2022, cl 19.31.31104**
  * Conan version: **conan 1.47.0**
  * Python version: **Python 3.10.2**


### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Windows
arch=x86_64
compiler=msvc
compiler.version=193
compiler.cppstd=17
compiler.runtime=dynamic
build_type=Release
[options]
[conf]
[build_requires]
[env]
```


### Steps to reproduce (Include if Applicable)

1. Freshly cloned cci repo
2. `conan create . autoconf/2.71@ -pr=windows-msvc-release-x86_64` inside `recipes/autoconf/all`

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
autoconf/2.71 (test package): Applying build-requirement: autoconf/2.71
autoconf/2.71 (test package): Applying build-requirement: msys2/cci.latest
autoconf/2.71 (test package): Applying build-requirement: m4/1.4.19
autoconf/2.71 (test package): Generator txt created conanbuildinfo.txt
autoconf/2.71 (test package): Aggregating env generators
autoconf/2.71 (test package): Generated conaninfo.txt
autoconf/2.71 (test package): Generated graphinfo
Using lockfile: 'C:\Data\Projects\conan-center-index\recipes\autoconf\all\test_package\build\186009612552ee0133c675206a8ad5077496a98d/conan.lock'
Using cached profile from lockfile
autoconf/2.71 (test package): Calling build()
autoconf/2.71 (test package): run_in_windows_bash: C:\conan\1cdc25\1\bin\msys64\usr\bin\bash.exe --login -c ^"cd \^"/c/data/projects/conan-center-index/recipes/autoconf/all/test_package/build/186009612552ee0133c675206a8ad5077496a98d\^" ^&^& PATH=\^"/c/conan/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin:/c/conan/1cdc25/1/bin/msys64/usr/bin:/c/conan/.conan/data/m4/1.4.19/_/_/package/0a420ff5c47119e668867cdb51baff0eca1fdb68/bin:$PATH\^" ^&^& /c/conan/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin/autoconf --verbose ^"
autoconf: running /c/conan/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin/autom4te --verbose --language=autoconf --output=configure C:\conan\.conan\data\autoconf\2.71\_\_\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\bin\share\autoconf/autoconf/trailer.m4 configure.ac
autom4te: the trace request object is:
autom4te: $VAR1 = bless( [
autom4te:                  '0',
autom4te:                  0,
autom4te:                  [
autom4te:                    'C:\\conan\\.conan\\data\\autoconf\\2.71\\_\\_\\package\\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\\bin\\share\\autoconf'
autom4te:                  ],
autom4te:                  [
autom4te:                    'C:/conan/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin/share/autoconf/autoconf/autoconf.m4f',
autom4te:                    'C:/conan/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin/share/autoconf/autoconf/trailer.m4',
autom4te:                    'configure.ac'
autom4te:                  ],
autom4te:                  {
autom4te:                    '_LT_AC_TAGCONFIG' => 1,
autom4te:                    'AC_CONFIG_FILES' => 1,
autom4te:                    'm4_pattern_allow' => 1,
autom4te:                    'AM_MAKEFILE_INCLUDE' => 1,
autom4te:                    'm4_sinclude' => 1,
autom4te:                    'AM_PROG_FC_C_O' => 1,
autom4te:                    '_m4_warn' => 1,
autom4te:                    'AM_EXTRA_RECURSIVE_TARGETS' => 1,
autom4te:                    'LT_CONFIG_LTDL_DIR' => 1,
autom4te:                    'AC_CANONICAL_SYSTEM' => 1,
autom4te:                    'LT_SUPPORTED_TAG' => 1,
autom4te:                    'AC_FC_PP_SRCEXT' => 1,
autom4te:                    'AM_PROG_MKDIR_P' => 1,
autom4te:                    '_AM_SUBST_NOTMAKE' => 1,
autom4te:                    'AM_PROG_CXX_C_O' => 1,
autom4te:                    'AC_REQUIRE_AUX_FILE' => 1,
autom4te:                    '_AM_COND_IF' => 1,
autom4te:                    'AM_CONDITIONAL' => 1,
autom4te:                    'AC_CONFIG_LINKS' => 1,
autom4te:                    'AC_FC_FREEFORM' => 1,
autom4te:                    'AC_SUBST' => 1,
autom4te:                    'AC_DEFINE_TRACE_LITERAL' => 1,
autom4te:                    'AM_PROG_F77_C_O' => 1,
autom4te:                    'AH_OUTPUT' => 1,
autom4te:                    'AC_CANONICAL_HOST' => 1,
autom4te:                    'AC_SUBST_TRACE' => 1,
autom4te:                    'AC_FC_SRCEXT' => 1,
autom4te:                    'AM_NLS' => 1,
autom4te:                    'AC_CONFIG_HEADERS' => 1,
autom4te:                    'AM_SILENT_RULES' => 1,
autom4te:                    'AM_INIT_AUTOMAKE' => 1,
autom4te:                    'AM_GNU_GETTEXT_INTL_SUBDIR' => 1,
autom4te:                    'AM_PROG_CC_C_O' => 1,
autom4te:                    'IT_PROG_INTLTOOL' => 1,
autom4te:                    'AM_PROG_AR' => 1,
autom4te:                    '_AM_COND_ENDIF' => 1,
autom4te:                    'include' => 1,
autom4te:                    'AC_CONFIG_LIBOBJ_DIR' => 1,
autom4te:                    'AC_CANONICAL_TARGET' => 1,
autom4te:                    'AM_PATH_GUILE' => 1,
autom4te:                    'AM_POT_TOOLS' => 1,
autom4te:                    'AC_LIBSOURCE' => 1,
autom4te:                    'AM_PROG_MOC' => 1,
autom4te:                    'AC_FC_PP_DEFINE' => 1,
autom4te:                    'AC_INIT' => 1,
autom4te:                    'AM_MAINTAINER_MODE' => 1,
autom4te:                    'AC_CONFIG_AUX_DIR' => 1,
autom4te:                    'm4_pattern_forbid' => 1,
autom4te:                    'AM_AUTOMAKE_VERSION' => 1,
autom4te:                    'm4_include' => 1,
autom4te:                    'AM_PROG_LIBTOOL' => 1,
autom4te:                    '_AM_COND_ELSE' => 1,
autom4te:                    'GTK_DOC_CHECK' => 1,
autom4te:                    'AM_ENABLE_MULTILIB' => 1,
autom4te:                    'AC_PROG_LIBTOOL' => 1,
autom4te:                    'AM_XGETTEXT_OPTION' => 1,
autom4te:                    'AC_CANONICAL_BUILD' => 1,
autom4te:                    'AC_CONFIG_SUBDIRS' => 1,
autom4te:                    'sinclude' => 1,
autom4te:                    'LT_INIT' => 1,
autom4te:                    'AC_CONFIG_MACRO_DIR_TRACE' => 1,
autom4te:                    '_AM_MAKEFILE_INCLUDE' => 1,
autom4te:                    'AM_GNU_GETTEXT' => 1
autom4te:                  }
autom4te:                ], 'Autom4te::Request' );
autom4te: running: C:/conan/.conan/data/m4/1.4.19/_/_/package/0a420ff5c47119e668867cdb51baff0eca1fdb68/bin/m4.exe --nesting-limit=1024 --gnu --include='C:\conan\.conan\data\autoconf\2.71\_\_\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\bin\share\autoconf' --debug=aflq --fatal-warning --debugfile=autom4te.cache/traces.0t --trace=AC_CANONICAL_BUILD --trace=AC_CANONICAL_HOST --trace=AC_CANONICAL_SYSTEM --trace=AC_CANONICAL_TARGET --trace=AC_CONFIG_AUX_DIR --trace=AC_CONFIG_FILES --trace=AC_CONFIG_HEADERS --trace=AC_CONFIG_LIBOBJ_DIR --trace=AC_CONFIG_LINKS --trace=AC_CONFIG_MACRO_DIR_TRACE --trace=AC_CONFIG_SUBDIRS --trace=AC_DEFINE_TRACE_LITERAL --trace=AC_FC_FREEFORM --trace=AC_FC_PP_DEFINE --trace=AC_FC_PP_SRCEXT --trace=AC_FC_SRCEXT --trace=AC_INIT --trace=AC_LIBSOURCE --trace=AC_PROG_LIBTOOL --trace=AC_REQUIRE_AUX_FILE --trace=AC_SUBST --trace=AC_SUBST_TRACE --trace=AH_OUTPUT --trace=AM_AUTOMAKE_VERSION --trace=AM_CONDITIONAL --trace=AM_ENABLE_MULTILIB --trace=AM_EXTRA_RECURSIVE_TARGETS --trace=AM_GNU_GETTEXT --trace=AM_GNU_GETTEXT_INTL_SUBDIR --trace=AM_INIT_AUTOMAKE --trace=AM_MAINTAINER_MODE --trace=AM_MAKEFILE_INCLUDE --trace=AM_NLS --trace=AM_PATH_GUILE --trace=AM_POT_TOOLS --trace=AM_PROG_AR --trace=AM_PROG_CC_C_O --trace=AM_PROG_CXX_C_O --trace=AM_PROG_F77_C_O --trace=AM_PROG_FC_C_O --trace=AM_PROG_LIBTOOL --trace=AM_PROG_MKDIR_P --trace=AM_PROG_MOC --trace=AM_SILENT_RULES --trace=AM_XGETTEXT_OPTION --trace=GTK_DOC_CHECK --trace=IT_PROG_INTLTOOL --trace=LT_CONFIG_LTDL_DIR --trace=LT_INIT --trace=LT_SUPPORTED_TAG --trace=_AM_COND_ELSE --trace=_AM_COND_ENDIF --trace=_AM_COND_IF --trace=_AM_MAKEFILE_INCLUDE --trace=_AM_SUBST_NOTMAKE --trace=_LT_AC_TAGCONFIG --trace=_m4_warn --trace=include --trace=m4_include --trace=m4_pattern_allow --trace=m4_pattern_forbid --trace=m4_sinclude --trace=sinclude --reload-state='C:/conan/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin/share/autoconf/autoconf/autoconf.m4f' --undefine=__m4_version__ 'C:/conan/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin/share/autoconf/autoconf/trailer.m4' configure.ac > autom4te.cache/output.0t
autom4te: formatting traces for '/tmp/am4t01vsl2/warnings': _m4_warn
autom4te: reading /tmp/am4t01vsl2/warnings
configure.ac:14: warning: The macro `AC_CONFIG_HEADER' is obsolete.
configure.ac:14: You should run autoupdate.
/c/conan/.conan/data/autoconf/2.71/_/_/build/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/source_subfolder/lib/autoconf/status.m4:719: AC_CONFIG_HEADER is expanded from...
configure.ac:14: the top level
autom4te: creating configure
autom4te: formatting traces for '/tmp/am4t01vsl2/patterns': m4_pattern_allow, m4_pattern_forbid
autom4te: forbidden tokens: ^_?A[CHUM]_|_AC_|^LIBOBJS$|^_?m4_|^dnl$|^_?AS_
autom4te: forbidden token : ^LIBOBJS$ => do not use LIBOBJS directly, use AC_LIBOBJ (see section `AC_LIBOBJ vs LIBOBJS'
autom4te: allowed   tokens: ^AS_FLAGS$|^SHELL$|^PATH_SEPARATOR$|^PACKAGE_NAME$|^PACKAGE_TARNAME$|^PACKAGE_VERSION$|^PACKAGE_STRING$|^PACKAGE_BUGREPORT$|^PACKAGE_URL$|^exec_prefix$|^prefix$|^program_transform_name$|^bindir$|^sbindir$|^libexecdir$|^datarootdir$|^datadir$|^sysconfdir$|^sharedstatedir$|^localstatedir$|^runstatedir$|^includedir$|^oldincludedir$|^docdir$|^infodir$|^htmldir$|^dvidir$|^pdfdir$|^psdir$|^libdir$|^localedir$|^mandir$|^PACKAGE_NAME$|^PACKAGE_TARNAME$|^PACKAGE_VERSION$|^PACKAGE_STRING$|^PACKAGE_BUGREPORT$|^PACKAGE_URL$|^DEFS$|^ECHO_C$|^ECHO_N$|^ECHO_T$|^LIBS$|^build_alias$|^host_alias$|^target_alias$|^CC$|^CFLAGS$|^LDFLAGS$|^LIBS$|^CPPFLAGS$|^CC$|^CC$|^CC$|^CC$|^ac_ct_CC$|^CC$|^EXEEXT$|^OBJEXT$|^CXX$|^CXXFLAGS$|^LDFLAGS$|^LIBS$|^CPPFLAGS$|^CXX$|^ac_ct_CXX$|^NO_MINUS_C_MINUS_O$|^CC_MINUS_O$|^CC_MINUS_O$|^LIBOBJS$|^LTLIBOBJS$
autom4te: 'configure' is created
autoconf/2.71 (test package): run_in_windows_bash: C:\conan\1cdc25\1\bin\msys64\usr\bin\bash.exe --login -c ^"cd \^"/c/data/projects/conan-center-index/recipes/autoconf/all/test_package/build/186009612552ee0133c675206a8ad5077496a98d\^" ^&^& PATH=\^"/c/conan/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin:/c/conan/1cdc25/1/bin/msys64/usr/bin:/c/conan/.conan/data/m4/1.4.19/_/_/package/0a420ff5c47119e668867cdb51baff0eca1fdb68/bin:$PATH\^" ^&^& C:/Data/Projects/conan-center-index/recipes/autoconf/all/test_package/build/186009612552ee0133c675206a8ad5077496a98d/configure --help ^"
`configure' configures test_package 1.0 to adapt to many kinds of systems.

Usage: C:/Data/Projects/conan-center-index/recipes/autoconf/all/test_package/build/186009612552ee0133c675206a8ad5077496a98d/configure [OPTION]... [VAR=VALUE]...

To assign environment variables (e.g., CC, CFLAGS...), specify them as
VAR=VALUE.  See below for descriptions of some of the useful variables.

Defaults for the options are specified in brackets.

Configuration:
  -h, --help              display this help and exit
      --help=short        display options specific to this package
      --help=recursive    display the short help of all the included packages
  -V, --version           display version information and exit
  -q, --quiet, --silent   do not print `checking ...' messages
      --cache-file=FILE   cache test results in FILE [disabled]
  -C, --config-cache      alias for `--cache-file=config.cache'
  -n, --no-create         do not create output files
      --srcdir=DIR        find the sources in DIR [configure dir or `..']

Installation directories:
  --prefix=PREFIX         install architecture-independent files in PREFIX
                          [/usr/local]
  --exec-prefix=EPREFIX   install architecture-dependent files in EPREFIX
                          [PREFIX]

By default, `make install' will install all the files in
`/usr/local/bin', `/usr/local/lib' etc.  You can specify
an installation prefix other than `/usr/local' using `--prefix',
for instance `--prefix=$HOME'.

For better control, use the options below.

Fine tuning of the installation directories:
  --bindir=DIR            user executables [EPREFIX/bin]
  --sbindir=DIR           system admin executables [EPREFIX/sbin]
  --libexecdir=DIR        program executables [EPREFIX/libexec]
  --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
  --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
  --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
  --runstatedir=DIR       modifiable per-process data [LOCALSTATEDIR/run]
  --libdir=DIR            object code libraries [EPREFIX/lib]
  --includedir=DIR        C header files [PREFIX/include]
  --oldincludedir=DIR     C header files for non-gcc [/usr/include]
  --datarootdir=DIR       read-only arch.-independent data root [PREFIX/share]
  --datadir=DIR           read-only architecture-independent data [DATAROOTDIR]
  --infodir=DIR           info documentation [DATAROOTDIR/info]
  --localedir=DIR         locale-dependent data [DATAROOTDIR/locale]
  --mandir=DIR            man documentation [DATAROOTDIR/man]
  --docdir=DIR            documentation root [DATAROOTDIR/doc/test_package]
  --htmldir=DIR           html documentation [DOCDIR]
  --dvidir=DIR            dvi documentation [DOCDIR]
  --pdfdir=DIR            pdf documentation [DOCDIR]
  --psdir=DIR             ps documentation [DOCDIR]

Some influential environment variables:
  CC          C compiler command
  CFLAGS      C compiler flags
  LDFLAGS     linker flags, e.g. -L<lib dir> if you have libraries in a
              nonstandard directory <lib dir>
  LIBS        libraries to pass to the linker, e.g. -l<library>
  CPPFLAGS    (Objective) C/C++ preprocessor flags, e.g. -I<include dir> if
              you have headers in a nonstandard directory <include dir>
  CXX         C++ compiler command
  CXXFLAGS    C++ compiler flags

Use these variables to override the choices made by `configure' or to help
it to find libraries and programs with nonstandard names/locations.

Report bugs to the package provider.
autoconf/2.71 (test package): run_in_windows_bash: C:\conan\1cdc25\1\bin\msys64\usr\bin\bash.exe --login -c ^"cd \^"/c/data/projects/conan-center-index/recipes/autoconf/all/test_package/build/186009612552ee0133c675206a8ad5077496a98d\^" ^&^& PATH=\^"/c/conan/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin:/c/conan/1cdc25/1/bin/msys64/usr/bin:/c/conan/.conan/data/m4/1.4.19/_/_/package/0a420ff5c47119e668867cdb51baff0eca1fdb68/bin:$PATH\^" ^&^& /c/data/projects/conan-center-index/recipes/autoconf/all/test_package/build/186009612552ee0133c675206a8ad5077496a98d/configure --help ^"
`configure' configures test_package 1.0 to adapt to many kinds of systems.

Usage: /c/data/projects/conan-center-index/recipes/autoconf/all/test_package/build/186009612552ee0133c675206a8ad5077496a98d/configure [OPTION]... [VAR=VALUE]...

To assign environment variables (e.g., CC, CFLAGS...), specify them as
VAR=VALUE.  See below for descriptions of some of the useful variables.

Defaults for the options are specified in brackets.

Configuration:
  -h, --help              display this help and exit
      --help=short        display options specific to this package
      --help=recursive    display the short help of all the included packages
  -V, --version           display version information and exit
  -q, --quiet, --silent   do not print `checking ...' messages
      --cache-file=FILE   cache test results in FILE [disabled]
  -C, --config-cache      alias for `--cache-file=config.cache'
  -n, --no-create         do not create output files
      --srcdir=DIR        find the sources in DIR [configure dir or `..']

Installation directories:
  --prefix=PREFIX         install architecture-independent files in PREFIX
                          [/usr/local]
  --exec-prefix=EPREFIX   install architecture-dependent files in EPREFIX
                          [PREFIX]

By default, `make install' will install all the files in
`/usr/local/bin', `/usr/local/lib' etc.  You can specify
an installation prefix other than `/usr/local' using `--prefix',
for instance `--prefix=$HOME'.

For better control, use the options below.

Fine tuning of the installation directories:
  --bindir=DIR            user executables [EPREFIX/bin]
  --sbindir=DIR           system admin executables [EPREFIX/sbin]
  --libexecdir=DIR        program executables [EPREFIX/libexec]
  --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
  --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
  --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
  --runstatedir=DIR       modifiable per-process data [LOCALSTATEDIR/run]
  --libdir=DIR            object code libraries [EPREFIX/lib]
  --includedir=DIR        C header files [PREFIX/include]
  --oldincludedir=DIR     C header files for non-gcc [/usr/include]
  --datadir=DIR           read-only architecture-independent data [DATAROOTDIR]
  --infodir=DIR           info documentation [DATAROOTDIR/info]
  --localedir=DIR         locale-dependent data [DATAROOTDIR/locale]
  --mandir=DIR            man documentation [DATAROOTDIR/man]
  --docdir=DIR            documentation root [DATAROOTDIR/doc/test_package]
  --htmldir=DIR           html documentation [DOCDIR]
  --dvidir=DIR            dvi documentation [DOCDIR]
  --pdfdir=DIR            pdf documentation [DOCDIR]
  --psdir=DIR             ps documentation [DOCDIR]

Some influential environment variables:
  CC          C compiler command
  CFLAGS      C compiler flags
  LDFLAGS     linker flags, e.g. -L<lib dir> if you have libraries in a
              nonstandard directory <lib dir>
  LIBS        libraries to pass to the linker, e.g. -l<library>
  CPPFLAGS    (Objective) C/C++ preprocessor flags, e.g. -I<include dir> if
              you have headers in a nonstandard directory <include dir>
  CXX         C++ compiler command
  CXXFLAGS    C++ compiler flags

Use these variables to override the choices made by `configure' or to help
it to find libraries and programs with nonstandard names/locations.

Report bugs to the package provider.
autoconf/2.71 (test package): Calling:
 > /c/data/projects/conan-center-index/recipes/autoconf/all/test_package/build/186009612552ee0133c675206a8ad5077496a98d/configure --prefix=C:/Data/Projects/conan-center-index/recipes/autoconf/all/test_package/build/186009612552ee0133c675206a8ad5077496a98d/package
autoconf/2.71 (test package): run_in_windows_bash: C:\conan\1cdc25\1\bin\msys64\usr\bin\bash.exe --login -c ^"cd \^"/c/data/projects/conan-center-index/recipes/autoconf/all/test_package/build/186009612552ee0133c675206a8ad5077496a98d\^" ^&^& PATH=\^"/c/conan/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin:/c/conan/1cdc25/1/bin/msys64/usr/bin:/c/conan/.conan/data/m4/1.4.19/_/_/package/0a420ff5c47119e668867cdb51baff0eca1fdb68/bin:$PATH\^" ^&^& /c/data/projects/conan-center-index/recipes/autoconf/all/test_package/build/186009612552ee0133c675206a8ad5077496a98d/configure --prefix=C:/Data/Projects/conan-center-index/recipes/autoconf/all/test_package/build/186009612552ee0133c675206a8ad5077496a98d/package  ^"
configure: loading site script /etc/config.site
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.exe
checking for suffix of executables... .exe
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking for g++... g++
checking whether the compiler supports GNU C++... no
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking whether gcc and cc understand -c and -o together... yes
configure: "ac_cv_prog_cc_gcc is gcc"
configure: creating ./config.status
config.status: creating Makefile
config.status: creating config.h
autoconf/2.71 (test package): run_in_windows_bash: C:\conan\1cdc25\1\bin\msys64\usr\bin\bash.exe --login -c ^"cd \^"/c/data/projects/conan-center-index/recipes/autoconf/all/test_package/build/186009612552ee0133c675206a8ad5077496a98d\^" ^&^& PATH=\^"/c/conan/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin:/c/conan/1cdc25/1/bin/msys64/usr/bin:/c/conan/.conan/data/m4/1.4.19/_/_/package/0a420ff5c47119e668867cdb51baff0eca1fdb68/bin:$PATH\^" ^&^& make -j24 ^"
gcc -dynamic -DNDEBUG -c test_package_c.c -o test_package_c.o
g++ -dynamic /std:c++17 -DNDEBUG -c test_package_cpp.cpp -o test_package_cpp.o
cc1: warning: unrecognized gcc debugging option: y
cc1: warning: unrecognized gcc debugging option: n
cc1: warning: unrecognized gcc debugging option: m
cc1: warning: unrecognized gcc debugging option: i
cc1: warning: unrecognized gcc debugging option: c
cc1plus: warning: unrecognized gcc debugging option: y
cc1plus: warning: unrecognized gcc debugging option: n
cc1plus: warning: unrecognized gcc debugging option: m
cc1plus: warning: unrecognized gcc debugging option: i
cc1plus: warning: unrecognized gcc debugging option: c
g++: warning: /std:c++17: linker input file unused because linking not done
g++: error: /std:c++17: linker input file not found: No such file or directory
make: *** [Makefile:11: test_package_cpp.o] Error 1
ERROR: autoconf/2.71 (test package): Error in build() method, line 42
        autotools.make()
        ConanException: Error 2 while executing make -j24
```

</details>

--- a/recipes/autoconf/all/test_package/conanfile.py
+++ b/recipes/autoconf/all/test_package/conanfile.py
@@ -1,4 +1,5 @@
 from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conan.tools.microsoft import is_msvc
 import contextlib
 import os
 import shutil
@@ -22,7 +23,7 @@ class TestPackageConan(ConanFile):
 
     @contextlib.contextmanager
     def _build_context(self):
-        if self.settings.compiler == "Visual Studio":
+        if is_msvc(self):
             with tools.vcvars(self):
                 with tools.environment_append({"CC": "cl -nologo", "CXX": "cl -nologo",}):
                     yield

--- a/recipes/autoconf/all/test_package/conanfile.py
+++ b/recipes/autoconf/all/test_package/conanfile.py
@@ -4,7 +4,7 @@ import contextlib
 import os
 import shutil
 
-required_conan_version = ">=1.36.0"
+required_conan_version = ">=1.45.0"
 
 
 class TestPackageConan(ConanFile):

